### PR TITLE
FOUNDATIONS: Upgrade User Modify

### LINKED ISSUE #90: FOUNDATIONS: Upgrade User Modify

### PR DESCRIPTION
**Pull request by GulchexrasGithub**
Closes #90

### CODE REVIEW

File: NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            var expectedUserDependencyValidationException =
+                new UserDependencyValidationException(lockedUserException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectUserByIdAsync(userId))
+                    .ThrowsAsync(databaseUpdateConcurrencyException);
+
+            // when
+            ValueTask<User> modifyUserTask =
+            this.userService.ModifyUserAsync(someUser);
```
Comment (StudentofTATU) [inline NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs:133]: tab

File: NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            UserDependencyValidationException actualUserDependencyValidationException =
```
Comment (StudentofTATU) [inline NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs:134]: empty line

File: NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            // given
+            int minutesInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDateTime = GetRandomDateTimeOffset();
+            User randomUser = CreateRandomUser(randomDateTime);
+            User someUser = randomUser;
+            Guid userId = someUser.Id;
+            var databaseUpdateException = new DbUpdateException();
+
+            var failedUserException =
+            new FailedUserStorageException(databaseUpdateException);
```
Comment (StudentofTATU) [inline NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Exceptions.Modify.cs:78]: tab

File: NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.Modify.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+            invalidUserException.AddData(
+                key: nameof(User.Email),
+                values: "Text is required");
+
+            invalidUserException.AddData(
+               key: nameof(User.LastName),
+               values: "Text is required");
+
+            invalidUserException.AddData(
```
Comment (StudentofTATU) [inline NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.Validations.Modify.cs:76]: order this FirstName should come before LastName as User.class written

File: NajotBooking.Api/Services/Foundations/Users/UserService.cs [hunk 5]
```diff
@@ -39,7 +39,17 @@ public partial class UserService : IUserService
             storageBroker.SelectAllUsers();
 
         public ValueTask<User> ModifyUserAsync(User user) =>
-            storageBroker.UpdateUserAsync(user);
+            TryCatch(async () =>
+            {
+                ValidateUser(user);
+
+                var maybeUser =
```
Comment (StudentofTATU) [inline NajotBooking.Api/Services/Foundations/Users/UserService.cs:46]: be more specific change var to User

**Review by StudentofTATU — COMMENTED** [review]
do some changes

File: NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.cs [hunk 6]
```diff
@@ -51,6 +57,13 @@ public UserServiceTests()
         private static Expression<Func<Xeption, bool>> SameExceptionAs(Xeption expectedException) =>
             actualException => actualException.SameExceptionAs(expectedException);
 
+        private static User CreateRandomModifyUser(DateTimeOffset date)
+        {
+            int randomDaysInPast = GetRandomNegativeNumber();
+            User randomUser = CreateRandomUser(date);
+            return randomUser;
```
Comment (Javlonbek-dev) [inline NajotBooking.Api.Tests.Unit/Services/Foundations/Users/UserServiceTests.cs:64]: An empty string is required before return

**Review by StudentofTATU — APPROVED** [review]

**Pull request merged**